### ISSUE TITLE
How to access the cache of the Rack Attack

### ISSUE THREAD

**Issue by nadavb** [kickstarter/rack-attack#217]
Let's say I want to create an internal page that displays the state of the rack attack, print all the throttles that reached their limits, all the bans, etc - how do I access this hash table information?

**Comment by maxwell** [comment]
```ruby
Rails.cache.data.keys.find_all{|x| x.include?('rack')}
```

works but isn't pretty.

**Comment by giedriusr** [comment]
@maxwell which cache backend are you using?

**Comment by motephyr** [comment]
Hi, I just use rack-attack in my project, it is very nice.
but I have a question, I don't know how to get throttle ip from Rails.cache
````
Rails.cache.data.keys.find_all{|x| x.include?('rack')}
````
but There is not data method or params.
How can I get ip list?

**Comment by courtenay** [comment]
I have a hack that does this:

```
class Rack::Attack
  LISTKEY = "rack::attack::ips"
   class Cache
      def write(unprefixed_key, value, expires_in)
        store.write("#{prefix}:#{unprefixed_key}", value, :expires_in => expires_in)
        if unprefixed_key =~ /fail2ban:(ban|count)/
          list = store.read(Rack::Attack::LISTKEY) || ""
          list = list.split(',')
          list << key
          list = list.uniq
          list = list.select {|k| !!client.get(k) }
          store.write(Rack::Attack::LISTKEY, list.uniq.join(','))
        end
      end
    end
```

**Comment by nadavb** [comment]
@gruzy, isn't the ActiveSupport::Notifications only for real time events? I'm looking for non-real-time, so in any moment you can view the state of cache(/memory) to understand all the accumulated data the rack attack has. For example, to see that for a given IP address, it used 500 out of the 1000 requests quota it has for this day.

**Comment by Kani999** [comment]
> ```ruby
> Rails.cache.data.keys.find_all{|x| x.include?('rack')}
> ```
> 
> works but isn't pretty.

I cannot access `data` 

```
Rails.cache.data
NoMethodError: undefined method `data' for #<ActiveSupport::Cache::FileStore:0x00000000072115b8>
```
Can you help me with this problem?

**Comment by lmansur** [comment]
I'm not really sure where this `data` method comes from. It doesn't look like its implemented by the default Rails stores.

**Comment by grzuy** [comment]
@Kani999 I think that only works if you're using a redis store.

Based on your comment you're using file store.

**Comment by Kani999** [comment]
> @Kani999 I think that only works if you're using a redis store.
> 
> Based on your comment you're using file store.

@grzuy 
Yes I'am using filestore. Is there some method, that I can list all blocked ip's? 
I just want to print out these.

**Comment by kaxla** [comment]
For anyone finding this by searching the error message `undefined method data' for #<ActiveSupport::Cache::RedisCacheStore:0x00007f8bbef18e10>` that results from running`Rails.cache.data` `data` has been renamed `redis` in Rails 5.2 so you have to do `Rails.cache.redis`

**Comment by matiasmasca** [comment]
> > ```ruby
> > Rails.cache.data.keys.find_all{|x| x.include?('rack')}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > works but isn't pretty.
> 
> I cannot access `data`
> 
> ```
> Rails.cache.data
> NoMethodError: undefined method `data' for #<ActiveSupport::Cache::FileStore:0x00000000072115b8>
> ```
> 
> Can you help me with this problem?

I was able to access in this way: **Rack::Attack**.cache.**store**
`Rack::Attack.cache.store.keys.find_all { |x| x.include?('rack::attack') }.sort`

**Comment by jacobluanjohnston** [comment]
I built a whole Rack::Attack dashboard if you want help, let me know. Problem is I can’t use any method (console, clearing redis) to lift a permanent ban. The cache might clear, but nothing can remove it (leading me to build a Rack::Attack dashboard to aide me in scouring every web resource out there).
